Makefile

```diff
@@ -292,3 +296,34 @@ clean:
 	rm -f $(TOOLBIN)/kustomize
 	rm -f $(TOOLBIN)/misspell
 	rm -f $(TOOLBIN)/mockgen
+
+
+## --------------------------------------
+## Releasing
+## --------------------------------------
+.PHONY: release-branch
+release-branch:
+	echo "checking branch=$(RELEASE_BRANCH)"
+	git ls-remote --exit-code `git remote get-url $(RELEASE_REMOTE)` $(RELEASE_BRANCH) || make create-release-branch
+
+.PHONY: create-release-branch
+create-release-branch:
+	git fetch upstream
+	git checkout master
+	git rebase upstream/master
+	git branch -D $(RELEASE_BRANCH) || true
+	git checkout -b $(RELEASE_BRANCH)
+	git push -f $(RELEASE_REMOTE) $(RELEASE_BRANCH)
+
+.PHONY: release-tag
+release-tag: release-branch
+	git branch -D $(RELEASE_BRANCH) || true
+	git branch ${RELEASE_BRANCH} ${RELEASE_REMOTE}/${RELEASE_BRANCH}
```

Comment:
It's confusing that we create the $(RELEASE_BRANCH) branch in the `create-release-branch`, and then delete the branch and fetch again from ${RELEASE_REMOTE}. Why are we doing that?

Reply:
I see the confusion. We cleanup the local ${RELEASE_BRANCH} and create a release branch pointing to the remote.